### DISCUSSION TITLE
use prisma client in the browser?

### DISCUSSION THREAD

**Discussion by andykais** [prisma/prisma#7403] [Q&A]
Hi, I am building a local web app. It has none of the normal concerns that go along with a typical `client <-> server` app. Prisma doesnt add a lot of value for me unless I have type safety across the client + server (since most server calls are just a pass through to sqlite database calls).

Is there any way to take advantage of the prisma query language & type safety inside a web browser? I imagine all that needs to happen is set up some kind of serializable message layer in the middle of the client which can be swapped out for an http server if it is meant to be used in the browser

**Comment by ryands17** [comment]
@andykais 👋 

Prisma currently cannot be used in the browser. There has to be a server layer that sends data to the browser. For e.g. In Next.js, you can use `getServerSideProps` or `getStaticProps` and call your Prisma API there where you'll get all the type safety. Another example is Electron where you can use IPC to send the results to the Web View and use the types directly there.


It would be great if you could add a 👍 to this request so that we can look at the priority: https://github.com/prisma/prisma/issues/4992

**Reply by andykais** [reply]
thanks for the info @ryands17. If I were using next.js or electron, those would be acceptable solutions. Unfortunately this is a svelte app. I will look into other rpc solutions as a go-between for now

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
